http-server/src/server.rs

```diff
@@ -113,26 +112,31 @@ pub struct Server {
 	/// Local address
 	local_addr: Option<SocketAddr>,
 	/// Registered methods.
-	root: RpcModule,
+	methods: Methods,
 	/// Max request body size.
 	max_request_body_size: u32,
 	/// Access control
 	access_control: AccessControl,
 }
 
 impl Server {
-	/// Register a new RPC method, which responds with a given callback.
-	pub fn register_method<F, R>(&mut self, method_name: &'static str, callback: F) -> Result<(), Error>
-	where
-		R: Serialize,
-		F: Fn(RpcParams) -> Result<R, CallError> + Send + Sync + 'static,
-	{
-		self.root.register_method(method_name, callback)
+	/// Register all [`Methods`] from an [`RpcModule`] on this server. In case a method already is registered with the same name,
+	/// the method is not added. Returns the number of methods registered on the server after the new module's methods
+	/// were added.
+	pub fn register_module<Context>(&mut self, module: RpcModule<Context>) -> usize {
```

Comment:
I would prefer to return `Error` here as it was before when duplicate methods are detected to make explicit for the user the handle the error.